src/cookies.rs

```diff
@@ -26,17 +26,18 @@ impl SecretKey {
 
 // Plugin boilerplate
 pub struct CookiePlugin;
-impl Key for CookiePlugin { type Value = CookieJar<'static>; }
+impl Key for CookiePlugin { type Value = CookieJar; }
 
 impl<'mw, 'conn, D> Plugin<Request<'mw, 'conn, D>> for CookiePlugin
 where D: KeyProvider {
     type Error = ();
 
-    fn eval(req: &mut Request<D>) -> Result<CookieJar<'static>, ()> {
-        let key = req.server_data().key();
+    fn eval(req: &mut Request<D>) -> Result<CookieJar, ()> {
         let jar = match req.origin.headers.get::<header::Cookie>() {
-            Some(c) => c.to_cookie_jar(&key.0),
-            None => CookieJar::new(&key.0)
+            //Some(c) => c.to_cookie_jar(&key.0),
+            Some(c) => c.iter().filter_map(|s| s.parse::<Cookie>().ok()).
```

Comment:
Can you remove the commented old code and knock this into it's own `{ block }` for ease of reading?

Reply:
Also maybe a note that malformed cookies are dropped here in the filter. (Seems fine to do that, perhaps it could go into a `debug!` log or something though incase someone did need to debug where their cookies went)

Reply:
Formatting changes in latest version of PR